facet-json/tests/option.rs

```diff
@@ -17,15 +17,48 @@ fn test_from_json_with_option() {
     }
 
     let json = r#"{
-    "name": "Alice",
-    "age": null,
-    "inner": {
-        "foo": 42
-    }
-}"#;
+        "name": "Alice",
+        "age": null,
+        "inner": {
+            "foo": 42
+        }
+    }"#;
 
     let test_struct: Options = from_str(json)?;
     assert_eq!(test_struct.name.as_deref(), Some("Alice"));
     assert_eq!(test_struct.age, None);
     assert_eq!(test_struct.inner.as_ref().map(|i| i.foo), Some(42));
 }
+
+#[test]
+fn test_from_json_with_nested_options() {
```

Comment:
Basically unrelated to this PR, but the changes in the deserializer incidentally made this work... although for some reason, setting `"inner": null` still fails to deserialize